vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListImagesController.java

```diff
@@ -65,4 +73,45 @@ public String listSpaces(@PathVariable(required = false) String page,
                 (order==null || order.equals("")) ? Sort.Direction.DESC.toString().toLowerCase():order);
         return "staff/images/imagelist";
     }
+    
+    @RequestMapping("/staff/images/list/tag/{filter}/{page}")
+    public String listSpacesByTag(@PathVariable(required = false) String page,
+            @PathVariable(required = false) String filter,
+            @RequestParam(value = "sort", required = false) String sortedBy,
+            @RequestParam(value = "order", required = false) String order, Model model) {
+        int pageNo;
+        page = StringUtils.isEmpty(page) ? "1" : page;
+        try {
+            pageNo = imageService.validatePageNumber(Integer.parseInt(page));
+        } catch (NumberFormatException numberFormatException){
+            pageNo = 1;
+        }
+        model.addAttribute("totalPages", imageService.getTotalPages());
+        model.addAttribute("currentPageNumber", pageNo);
+        model.addAttribute("totalImageCount", imageService.getTotalImageCount());
+        List<IVSImage> imagesList = imageService.getImages(pageNo, sortedBy, order);
```

Comment:
you should only get the images from the db that have a certain category (your total image count is wrong here too, isn't it?). Also, this should probably happen in the `listSpaces` method and pass the filter as query parameters. A lot of duplicate code here.